neotest-kotlin/app/src/test/kotlin/io/github/codymikol/neotestkotlin/framework/kotest/KotestTestRunnerFunctionalSpec.kt

```diff
@@ -0,0 +1,164 @@
+package io.github.codymikol.neotestkotlin.framework.kotest
+
+import io.github.codymikol.neotestkotlin.framework.TestRunResult
+import io.kotest.assertions.json.shouldEqualSpecifiedJson
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.types.shouldBeInstanceOf
+import kotlinx.serialization.json.Json
+
+class KotestTestRunnerFunctionalSpec :
+    FunSpec({
+        context("functional") {
+            test("run") {
+                val result = KotestTestRunner.run(listOf(KotestExampleSpec::class))
+                val actual = result.shouldBeInstanceOf<TestRunResult.Success>()
+                val actualJson = Json.encodeToString(actual.report)
+                actualJson shouldEqualSpecifiedJson
+                    """
+                    [
+                      {
+                        "name": "io.github.codymikol.neotestkotlin.framework.kotest.KotestExampleSpec",
+                        "nodes": [
```

Comment:
Don't love the "nodes" naming here, but we effectively have a tree. I also don't know a good word for either a Container or a Test